File: src/main/java/com/basecolon/uisless/UISless.java
package com.basecolon.uisless;

import com.basecolon.uisless.database.Database;
import com.basecolon.uisless.model.ConfigModel;
import com.basecolon.uisless.model.CourseModel;
import com.basecolon.uisless.scraper.CourseScraper;
import com.google.gson.Gson;

import java.io.FileNotFoundException;
import java.io.InputStream;
import java.util.List;
import java.util.Scanner;

public class UISless {
    private static final String CONFIG_FILE_NAME = "sensitive_config.json";

    public static void main(String[] args) throws FileNotFoundException {
        //Load the config file into memory
        final ClassLoader classloader = Thread.currentThread().getContextClassLoader();
        final InputStream inputStream = classloader.getResourceAsStream(CONFIG_FILE_NAME);

        //Convert config into a Java class
        final String configString = convertStreamToString(inputStream);
        final ConfigModel config =  new Gson().fromJson(configString, ConfigModel.class);

        final Database database = Database.getInstance(config.database, config.term);

        final CourseScraper scraper = new CourseScraper(config.user, config.term);
        final List<CourseModel> courses = scraper.getClassRecords();

        //Insert every course model into the DB
        courses.forEach(database::insert);
    }

    public static String convertStreamToString(InputStream is) {
        Scanner s = new Scanner(is).useDelimiter("\\A");
        return s.hasNext() ? s.next() : "";
    }
}

File: src/main/java/com/basecolon/uisless/UISless.kt
package com.basecolon.uisless

import com.basecolon.uisless.database.Database
import com.basecolon.uisless.model.ConfigModel
import com.basecolon.uisless.scraper.CourseScraper
import com.google.gson.Gson
import java.io.InputStream
import java.util.*

object UISless {
    private val CONFIG_FILE_NAME = "sensitive_config.json"

    fun main(args: Array<String>) {
        //Load the config file into memory
        val classloader = Thread.currentThread().contextClassLoader
        val inputStream = classloader.getResourceAsStream(CONFIG_FILE_NAME)

        //Convert config into a Java class
        val configString = convertStreamToString(inputStream)
        val config = Gson().fromJson<ConfigModel>(configString, ConfigModel::class.java!!)

        val database = Database.getInstance(config.database, config.term)

        val scraper = CourseScraper(config.user, config.term)
        val courses = scraper.classRecords

        //Insert every course model into the DB
        courses.forEach({ database.insert(it) })
    }

    private fun convertStreamToString(`is`: InputStream): String {
        val s = Scanner(`is`).useDelimiter("\\A")
        return if (s.hasNext()) s.next() else ""
    }
}
